csi-plug-n-play/sidecars/soda-csi-provisioner/external-provisioner/pkg/controller/controller.go

```diff
@@ -494,13 +499,56 @@ type prepareProvisionResult struct {
 	req            *csi.CreateVolumeRequest
 	csiPVSource    *v1.CSIPersistentVolumeSource
 }
+type CustomPropertiesSpec map[string]interface{}
+
+func (cps CustomPropertiesSpec) GetDriverPreference() string {
+	var driverName string
+	for k, v := range cps {
+		if k == "driver" {
+			driverName = fmt.Sprintf("%v", v)
+		}
+	}
+	return driverName
+}
 
 // prepareProvision does non-destructive parameter checking and preparations for provisioning a volume.
 func (p *csiProvisioner) prepareProvision(ctx context.Context, claim *v1.PersistentVolumeClaim, sc *storagev1.StorageClass, selectedNode *v1.Node) (*prepareProvisionResult, controller.ProvisioningState, error) {
 	if sc == nil {
 		return nil, controller.ProvisioningFinished, errors.New("storage class was nil")
 	}
 
+	//SODA-CHANGE
+	backendDriverName, err := GetDriverName(p.grpcClient, operationTimeout)
+	if err != nil {
+		klog.Fatalf("Error getting CSI driver name: %s", err)
+	}
+	klog.Infof("The Backend Driver Name and provisioner name is : %s , %s", backendDriverName, p.driverName)
+
+	if sc.Provisioner == "soda-csi" {
+		for k, v := range sc.Parameters {
+			klog.Infof("The parameters in the StorageClass are  : %s ===== %s", k, v)
+			if k == "profile" {
+
+				response, err := http.Get("http://soda-proxy:50029/getprofile/" + v)
+				if err != nil {
+					klog.Infof("Got error in GetProfile  : %s ===== %s", err.Error())
+				} else {
+					data, _ := ioutil.ReadAll(response.Body)
+					var customProperties *CustomPropertiesSpec
+					json.Unmarshal(data, &customProperties)
+					klog.Infof("The profile name received in the storageClass is: %s", customProperties.GetDriverPreference())
+					if backendDriverName != customProperties.GetDriverPreference() {
```

Comment:
multiple calls to customProperties.GetDriverPreference() can be avoided for line 539 and 540